src/components/LoginForm.tsx

```diff
@@ -28,12 +47,219 @@ export default function LoginForm({
+                user.tg = tg;
+                user.sigPk = pk;
+                user.msg = msg;
+                user.sig = sig;
+                user.inTs = metadata.timestamp;
+
+                users[userId] = user;
+              } else {
+                users[userId] = {
+                  name: metadata.fromDisplayName,
+                  encPk: metadata.fromPublicKey,
+                  x,
+                  tg,
+                  sigPk: pk,
+                  msg,
+                  sig,
+                  inTs: metadata.timestamp,
+                };
+              }
+
+              const activity = {
+                type: JUB_SIGNAL_MESSAGE_TYPE.INBOUND_TAP,
+                name: metadata.fromDisplayName,
+                id: userId,
+                ts: metadata.timestamp,
+              };
+              activities.push(activity);
+            } catch (error) {
+              console.error(
+                "Invalid inbound tap message received from server: ",
+                message
+              );
+            } finally {
+              break;
+            }
+          case JUB_SIGNAL_MESSAGE_TYPE.LOCATION_TAP:
+            try {
+              if (metadata.fromPublicKey !== profile.encryptionPublicKey) {
+                throw new Error(
+                  "Invalid message: location tap messages must be sent from self"
+                );
+              }
+
+              const { id, name, pk, msg, sig } =
+                await locationTapMessageSchema.validate(data);
+              const location = locationSignatures[id];
+              if (location) {
+                location.id = id;
+                location.name = name;
+                location.pk = pk;
+                location.msg = msg;
+                location.sig = sig;
+                location.ts = metadata.timestamp;
+
+                locationSignatures[id] = location;
+              } else {
+                locationSignatures[id] = {
+                  id,
+                  name,
+                  pk,
+                  msg,
+                  sig,
+                  ts: metadata.timestamp,
+                };
+              }
+
+              const activity = {
+                type: JUB_SIGNAL_MESSAGE_TYPE.LOCATION_TAP,
+                name,
+                id,
+                ts: metadata.timestamp,
+              };
+              activities.push(activity);
+            } catch (error) {
+              console.error(
+                "Invalid location tap message received from server: ",
+                message
+              );
+            } finally {
+              break;
+            }
+          case JUB_SIGNAL_MESSAGE_TYPE.QUEST_COMPLETED:
+            throw new Error("Unable to handle quest completed messages");
+          default:
+            throw new Error("Invalid message type");
+        }
+      } catch (error) {
+        console.error("Invalid message received from server: ", message);
+      }
+    });
+
```

Comment:
In general, this entire section doesn't really have any real error catching. Just console errors. We should have some sort of error screen to send users to, as well as info on who to contact.

Reply:
This applies to pretty much all the flows we have now - adding a ticket here: https://github.com/nfc-denver/nfc-denver/issues/67